hbase_sql/src/main/java/com/splicemachine/compactions/SpliceCompactionRequest.java

```diff
@@ -51,19 +55,35 @@ public void preStorefilesRename() throws IOException {
                 }
                 continue;
             }
-            if(memstoreAware.compareAndSet(latest, MemstoreAware.incrementCompactionCount(latest)))
+            if(memstoreAware.compareAndSet(latest, MemstoreAware.incrementCompactionCount(latest))) {
+                if(LOG.isDebugEnabled()) {
+                    SpliceLogUtils.debug(LOG, "memstoreAware@" + System.identityHashCode(memstoreAware) +
+                            " 's compactionCount incremented from " + latest.currentCompactionCount +
+                            " to " + (latest.currentCompactionCount + 1));
+                }
+                assert !compactionCountIncremented;
+                compactionCountIncremented = true;
                 break;
+            }
         }
     }
     public void afterExecute(){
-        if (memstoreAware == null) {
+        if (memstoreAware == null || !compactionCountIncremented) {
```

Comment:
I don't think so. Without `compactionCountIncremented`, we end up decrementing compactionCount below 0.
That's because we setMemstoreAware in preCompact but we call preStorefilesRename after the actual execution in SpliceDefaultCompactor.compact()

Reply:
That's right, I got confused and thought we were setting it in preStorefilesRename. Looks good!